pkg/rtc/participant.go

```diff
@@ -963,133 +868,24 @@ func (p *ParticipantImpl) VerifySubscribeParticipantInfo(pID livekit.Participant
 	}
 }
 
-// AddSubscribedTrack adds a track to the participant's subscribed list
-func (p *ParticipantImpl) AddSubscribedTrack(subTrack types.SubscribedTrack, sourceTrack types.MediaTrack) {
```

Comment:
reduced responsibilities of this function, and changed to onTrackSubscribed as a callback